catalog/imaging/imaging-postprocess-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,556 @@
+
+    return render(
+        inputStream,
+        input -> {
+          try {
+            return input.getRight().render(input.getLeft());
+          } catch (IOException e) {
+            LOGGER.debug(e.getMessage(), e);
+          }
+          return null;
+        });
+  }
+
+  private BufferedImage renderImageUsingOriginalDataModel(InputStream inputStream)
+      throws IOException, NitfFormatException, InterruptedException {
+
+    return render(
+        inputStream,
+        input -> {
+          try {
+            return input.getRight().renderToClosestDataModel(input.getLeft());
+          } catch (IOException e) {
+            LOGGER.debug(e.getMessage(), e);
+          }
+          return null;
+        });
+  }
+
+  private BufferedImage render(
+      InputStream inputStream,
+      Function<Pair<ImageSegment, NitfRenderer>, BufferedImage> imageSegmentFunction)
+      throws IOException, NitfFormatException, InterruptedException {
+
+    available.acquire();
+    final ThreadLocal<BufferedImage> bufferedImage = new ThreadLocal<>();
+
+    if (inputStream != null) {
+      try {
+        NitfRenderer renderer = nitfRendererSupplier.get();
+
+        new NitfParserInputFlowImpl()
+            .inputStream(inputStream)
+            .allData()
+            .forEachImageSegment(
+                segment -> {
+                  if (bufferedImage.get() == null) {
+                    BufferedImage bi =
+                        imageSegmentFunction.apply(new ImmutablePair<>(segment, renderer));
+                    if (bi != null) {
+                      bufferedImage.set(bi);
+                    }
+                  }
+                })
+            .end();
+      } finally {
+        IOUtils.closeQuietly(inputStream);
+        available.release();
+      }
+    }
+
+    return bufferedImage.get();
+  }
+
+  private void addThumbnailToMetacard(Metacard metacard, BufferedImage bufferedImage) {
+    try {
+      byte[] thumbnailImage = scaleImage(bufferedImage, THUMBNAIL_WIDTH, THUMBNAIL_HEIGHT);
+
+      if (thumbnailImage.length > 0) {
+        metacard.setAttribute(new AttributeImpl(Core.THUMBNAIL, thumbnailImage));
+      }
+    } catch (IOException e) {
+      LOGGER.debug(e.getMessage(), e);
+    }
+  }
+
+  private ProcessResource createDerivedImage(
+      String qualifier, BufferedImage image, Metacard metacard, int maxWidth, int maxHeight) {
+    try {
+      byte[] overviewBytes = scaleImage(image, maxWidth, maxHeight);
+      InputStream overviewBytesInputStream = new ByteArrayInputStream(overviewBytes);
+
+      ProcessResource processResource =
+          new ProcessResourceImpl(
+              metacard.getId(),
+              overviewBytesInputStream,
+              IMAGE_JPEG,
+              buildDerivedImageTitle(metacard.getTitle(), qualifier, JPG),
+              overviewBytes.length,
+              qualifier);
+
+      ((ProcessResourceImpl) processResource).markAsModified();
```

Comment:
✏️ Since `processResource` is a local variable, should we just declare it to be a `ProcessResourceImpl` and avoid the cast?